server/src/main/java/org/elasticsearch/health/HealthStatus.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.health;
+
+import java.util.stream.Stream;
+
+public enum HealthStatus {
+    GREEN,
+    YELLOW,
+    RED;
+
+    public static HealthStatus merge(Stream<HealthStatus> statuses) {
+        return statuses.max(HealthStatus::compareTo).orElse(GREEN);
```

Comment:
Consider using explicit ordering instead of relying on `Enum.ordinal` which can change if the enum entries get reordered by accident (e.g. automatically by a code formatter)

See _Effecitve Java Item 35: Use instance fields instead of ordinals_: 
https://github.com/nicolasmanic/effective-java-in-a-nutshell/blob/master/README.md#item-35-use-instance-fields-instead-of-ordinals